src/Components/PaperEditor/TranscriptsContainer/TranscriptTabContent/index.js

```diff
@@ -116,49 +110,215 @@ const TranscriptTabContent = (props) => {
-    const cssName = words.join(' ');
-    const paragraphCSS = `.paragraph[data-paragraph-text*="${ cssName }"]`;
-
-    const css = words.reduce(
-      (res, word) => {
-        res.paragraphs.push(
-          `${ paragraphCSS } > div > span.words[data-text="${ word }"]`
-        );
-        res.search.push(
-          `${ paragraphCSS } > div > span >span.words[data-text="${ word }"]`
-        );
-
-        return res;
-      },
-      { paragraphs: [], search: [] }
-    );
-    const wordsToSearchCSS = css.paragraphs.join(', ');
-    // Need to add an extra span to search annotation hilights
-    // TODO: refactor to make more DRY
-    const wordsToSearchCSSInHighlights = css.search.join(', ');
-    setSentenceToSearchCSS(wordsToSearchCSS);
-    setSentenceToSearchCSSInHighlights(wordsToSearchCSSInHighlights);
+    return () => {};
+  }, [ transcript ]);
+
+  useEffect(() => {
+    if (
+      searchString !== '' ||
+      selectedSpeakers.length > 0 ||
+      selectedLabels.length > 0
+    ) {
+      setHasSearch(true);
+    } else {
+      setHasSearch(false);
+    }
+
+    return () => {
+      setHasSearch(false);
+    };
+  }, [ searchString, selectedLabels, selectedSpeakers ]);
+
+  useEffect(() => {
+    /* Paragraph text for data attribute for searches, without punctuation */
+    const findString = (str, text) => (str ? text.includes(str) : false);
+
+    const findSpeaker = (speaker) =>
+      !!selectedSpeakers.find((spk) => spk.label === speaker);
+
+    const findLabel = (label) => !!selectedLabels.find((lb) => lb.id === label);
+
+    const isSearchResult = (text, speaker, label) => {
+      const textWithoutPunctuation = removePunctuation(text);
+      const lcSearchString = searchString.toLowerCase();
+
+      const foundSearchString = findString(
+        lcSearchString,
+        textWithoutPunctuation
+      );
+
+      const foundSpeaker = findSpeaker(speaker, selectedSpeakers);
+      const foundLabel = findLabel(label, selectedLabels);
+
+      if (foundSearchString || foundSpeaker || foundLabel) {
+        return true;
+      }
+
+      return false;
+    };
+
+    const isParagraphSearchResult = (paragraph) => {
+      let wordsAnnotations = { labelId: '' };
+      if (annotations) {
+        wordsAnnotations = findAnnotationsInWords(annotations, paragraph.words);
+      }
+
+      if (
+        isSearchResult(
+          paragraph.text,
+          paragraph.speaker,
+          wordsAnnotations.labelId
+        )
+      ) {
+        return true;
+      }
+
+      return false;
+    };
+
+    const setAllDisplays = (display) => {
+      const displayAllParagraphs = paragraphs.map((p) => display);
+      setDisplayParagraphs(displayAllParagraphs);
+    };
+    const setAllSearchResults = (searchResult) => {
+      const searchResults = paragraphs.map((p) => searchResult);
+      setIsSearchResults(searchResults);
+    };
+
+    if (paragraphs) {
+      if (paragraphOnly) {
```

Comment:
`paragraphOnly` shows only matching paragraphs when searching. It's the checkbox in the search interface.